### DISCUSSION TITLE
Bulk install of apps with home-brew on new system (Mac OS 11.2.3 Big Sur)

### DISCUSSION THREAD

**Discussion by steve-s1w** [Homebrew/discussions#1239] [Casks]
I have a new Mac and want to use home-brew to install as many apps as possible (on my old Mac I just used the install cask command individually).

Is there any way I can just put a list of apps into home-brew and it'll just go ahead and install them all without my interaction?  This will be used on an M1 Mac mini.  This is a list of apps:  Thanks all

Ableton Live 10 Suite
adbLink
Adobe Audition CC 2019
Affinity Designer
Affinity Photo
Any Video Converter
App Store
AppCleaner
Audio Hijack
Audiofinder
Authorizer
Authy Desktop
Automator
balenaEtcher
Bear
Books
Brave Browser
BUTT
Calculator
Calendar
calibre
Camera Settings
Carbon Copy Cloner
Chess
Contacts
Cubase 10.5
Cyberduck
Dictionary
Discord
djay Pro 2
djay Pro AI
draw.io
Dropbox
eLicenser Control Center
eSPi
FaceTime
FileBot
Find My
Firefox
Font Book
GarageBand
GhostReader
Google Chrome
GrandPerspective
HandBrake
HideSwitch
Home
Image Capture
iMovie
Install macOS Big Sur
iStat Menus
iVMS-4200
Keka
Keynote
kid3
LanScan
LastPass
Launchpad
Little Snitch
Logi Capture
Logic Pro
Luxriot EVO
Macs Fan Control
Mactracker
Mail
Maps
MediaInfo
Messages
MickTagger
Microsoft Excel
Microsoft OneNote
Microsoft Outlook
Microsoft PowerPoint
Microsoft Teams
Microsoft Word
MIDI Monitor
Mission Control
MKVToolNix-45.0.0
Music
Native Access
Native Instruments
News
Newshosting
Notes
Numbers
OBS
OneDrive
OneNote Importer (Preview)
owncloud
Pages
Panda
Photo Booth
Photos
Plex
Podcasts
Preview
PureVPN
qBittorrent
QuickTime Player
Raspberry Pi Imager
Reason 10
Reminders
Remote Desktop
SABnzbd
SADPTool
Safari
Sensei
Serato DJ Pro
Signal
Siri
SiteSucker
Slack
SoulseekQt
Spark
Spotify
Steinberg Download Assistant
Steinberg Library Manager
Stickies
Stocks
Swinsian
System Preferences
Tapo
TeamViewer
Telegram
TextEdit
Textual
Textual 7
The Unarchiver
Time Machine
TinkerTool
Trello
TuneIn
TV
UniFi
UniFi-Discover
Utilities
VLC
Voice Memos
Vonage Business
Wayback Machine
WhatsApp
YACReader
YACReaderLibrary

**Comment by carlocab** [comment]
Convert your list into a list of names of formulae and casks. Then run
```
brew install $(cat list.txt)
```
where `list.txt` is the filename of your list saved in the current directory. If it's not in the current directory, just specify the full path.

**Comment by SMillerDev** [comment]
This sounds like exactly what https://github.com/Homebrew/homebrew-bundle does.

**Comment by FredrikWendt** [comment]
I use Ansible to automate things like this. I have a playbook that lists all the packages and just run one command to install it all.

**Comment by mmoole** [comment]
recently I also came across this, and there are mainly those ways to go - all of them need the cask or package names as input:
* ansible as mentioned, maybe with a custom config, via https://github.com/geerlingguy/mac-dev-playbook
* brew bundle as mentioned
* brew in a loop, as mentioned there: https://apple.stackexchange.com/questions/279077/how-can-i-install-all-brew-packages-listed-in-a-text-file
  * i did it via `for i in $(cat brew_installed_cask.txt); do; brew install --cask "$i"; done`
